app/models/user.rb

```diff
@@ -87,20 +85,38 @@ def self.next_membership_payment_date(user_id)
     next_membership_payment_dates(user_id).first
   end
 
-
   # TODO this should not be the responsibility of the User class. Need a MembershipManager class for this.
   def self.next_membership_payment_dates(user_id)
     next_payment_dates(user_id, THIS_PAYMENT_TYPE)
   end
 
-
-  # ----------------------------------
+  def self.clear_all_proof_of_membership_jpg_caches
+    all.each do |user|
+      user.clear_proof_of_membership_jpg_cache
+    end
+  end
 
   after_update :clear_proof_of_membership_jpg_cache,
                if: Proc.new { saved_change_to_member_photo_file_name? ||
-                              saved_change_to_first_name? ||
-                              saved_change_to_last_name? ||
-                              saved_change_to_membership_number? }
+                 saved_change_to_first_name? ||
+                 saved_change_to_last_name? ||
+                 saved_change_to_membership_number? }
+
+  # @return [ActiveSupport::Duration]
+  def self.membership_expired_grace_period
+    ActiveSupport::Duration.days(AdminOnly::AppConfiguration.config_to_use.membership_expired_grace_period.to_i)
```

Comment:
Couldn't this be simplified to:
```
AdminOnly::AppConfiguration.config_to_use.membership_expired_grace_period.days
```
?

Reply:
doh!  I somehow totally missed that an Integer can be converted to a Duration.
Yes, that will simplify things.